crates/starknet/src/starknet/mod.rs

```diff
@@ -163,6 +163,7 @@ impl Starknet {
     // Initialize values for new pending block
     pub(crate) fn generate_pending_block(&mut self) -> DevnetResult<()> {
         Self::update_block_context(&mut self.block_context);
+        self.update_pending_block_timestamp();
```

Comment:
This method is called in four places in the code. Could it be better encapsulated somewhere?

Reply:
I already tried that but it causing problems with other parts of the code. So far is the best solution that I could find without breaking normal devnet workflow.